### PR TITLE
[Rocm] fix expand as

### PR DESCRIPTION
**Pull request by Ray2020BD**
### PR types

Bug fixes
### PR changes

APIs
### Describe

[Rocm] fix test_expand_as_op

Add eigen-double support with rocm.

image

### CODE REVIEW

File: cmake/external/eigen.cmake [hunk 1]
```diff
@@ -33,6 +33,8 @@ elseif(LINUX)
         # which will cause compiler error of using __host__ funciont in __host__ __device__
         file(TO_NATIVE_PATH ${PADDLE_SOURCE_DIR}/patches/eigen/Meta.h native_src)
         file(TO_NATIVE_PATH ${EIGEN_SOURCE_DIR}/Eigen/src/Core/util/Meta.h native_dst)
+        file(TO_NATIVE_PATH ${PADDLE_SOURCE_DIR}/patches/eigen/TensorReductionGpu.h native_src)
+        file(TO_NATIVE_PATH ${EIGEN_SOURCE_DIR}/unsupported/Eigen/CXX11/src/Tensor/TensorReductionGpu.h native_dst)
```
Comment (qili93) [inline cmake/external/eigen.cmake]: 这里修改的不对，会把上一个的patch覆盖掉了，当存在多个patch的时候，需要参考这个修改 https://github.com/qili93/Paddle/blob/rocm_dev_0207/cmake/external/eigen.cmake#L61

**Review by qili93 — CHANGES_REQUESTED** [review]

File: cmake/external/eigen.cmake (hunk 1 above)
Comment (Ray2020BD) [reply cmake/external/eigen.cmake]: done.

**Review by qili93 — APPROVED** [review]
LGTM，再给 expand_as op的负责人帮忙看一下有没有问题

**Comment by HexToString** [comment]
LGTM

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**